views/FanView/FanView.tsx

```diff
@@ -74,31 +124,57 @@ export const FanView: FC<FanViewProps> = ({ route }) => {
   return (
     <View style={styles.container}>
       <Navbar text={`${location} - ${nodeId} - ${fan?.name}`} />
-      <View style={styles.content}>
-        <Typography
-          variant="body-medium"
-          text={`Aktualny temperatura wentylacji: ${fan.values.V_TEMP}`}
-        />
-        <Input text={temp} onChange={onTempChange} />
-        <Typography
-          variant="body-medium"
-          text={`Aktualny prędkość wentylacji: ${fan.values.V_DIRECTION}`}
-        />
-        <Input text={direction} onChange={onDirectionChange} />
-        <Typography
-          variant="body-medium"
-          text={`Aktualny kierunek głowicy wentylacji: ${fan.values.V_PERCENTAGE}`}
-        />
-        <Input text={percentage} onChange={onPercentageChange} />
-        <Button
-          text="Zmień parametry wentylatora"
-          onPress={handleChangeFanParams}
-        />
-        <Button
-          text="Pobierz najnowsze wartości"
-          onPress={handleGetFanParams}
-        />
-      </View>
+      <ScrollView>
+        <View style={styles.content}>
+          <View style={styles.section}>
+            <Typography
+              variant="body-medium"
+              text={`Aktualna temperatura wentylacji: ${fan.values.V_TEMP}°C`}
+            />
+            <Input
+              text={temp}
+              onChange={e => {
+                checkInput(e, onTempChange, 0, 50);
```

Comment:
Instead of passing `checkInput` as a onChange function, move the code that is checking if input is correct and run it automatically if `keyboardType` is equal to numeric. Also, a min and a max should be a prop (similar to input type="numeric" in HTML)

Reply:
Done